### PR TITLE
Update Azure AI Search (fka Azure Cognitive Search) vector store integration to latest client SDK 11.4.0 stable + updating jupyter notebook sample 

### PR DESCRIPTION
**Pull request by farzad528**
# Description

This PR includes updates to Azure AI Search (renamed from Azure Cognitive Search) vector store integration to the latest azure-search-documents pypi stable release version along with minor syntax updates. The jupyter notebook is also updated. 

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by farzad528** [comment]
@pablocastro @logan-markewich plz review

**Comment by logan-markewich** [comment]
@farzad528 Thank you! Will get this reviewed today :)

File: llama_index/vector_stores/azureaisearch.py [hunk 1]
```diff
@@ -141,68 +148,69 @@ def _create_index(self, index_name: Optional[str]) -> None:
...
                 name=self._field_mapping["embedding"],
                 type=SearchFieldDataType.Collection(SearchFieldDataType.Single),
-                hidden=False,
                 searchable=True,
-                filterable=False,
-                sortable=False,
-                facetable=False,
                 vector_search_dimensions=self.embedding_dimensionality,
-                vector_search_configuration="default",
+                vector_search_profile_name="myHnswProfile",
```
Comment (hatianzhang) [inline llama_index/vector_stores/azureaisearch.py:153]: does the default profile name need to be like this

File: llama_index/vector_stores/azureaisearch.py [hunk 2]
```diff
@@ -141,68 +148,69 @@ def _create_index(self, index_name: Optional[str]) -> None:
...
-                    kind="hnsw",
-                    parameters=hnsw_param,
-                )
-            ]
+            algorithms=[
+                HnswAlgorithmConfiguration(
+                    name="myHnsw",
+                    kind=VectorSearchAlgorithmKind.HNSW,
+                    parameters=HnswParameters(
+                        m=4,
```
Comment (hatianzhang) [inline llama_index/vector_stores/azureaisearch.py:171]: might be useful to link the doc for those HNSW parameters

File: llama_index/vector_stores/azureaisearch.py (hunk 1 above)
Comment (farzad528) [reply llama_index/vector_stores/azureaisearch.py:153]: no, it can be named anything

Comment (hatianzhang) [reply llama_index/vector_stores/azureaisearch.py:153]: could we just use `default`

File: docs/examples/vector_stores/AzureAISearchIndexDemo.ipynb [hunk 3]
```diff
@@ -0,0 +1,730 @@
...
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "hybrid_retriever = index.as_retriever(\n",
+    "    vector_store_query_mode=VectorStoreQueryMode.SEMANTIC_HYBRID\n",
+    ")\n",
+    "hybrid_retriever.retrieve(\"What is inception about?\")\n",
+    "\n",
+    "hybrid_retriever.retrieve(\"What is inception about?\")"
```
Comment (pablocastro) [inline docs/examples/vector_stores/AzureAISearchIndexDemo.ipynb:705]: Did you mean to repeat this line?

**Review by pablocastro — COMMENTED** [review]
Looks great overall, left a few comments on smaller details.

File: pyproject.toml [hunk 4]
```diff
@@ -77,6 +77,10 @@ scikit-learn = {optional = true, version = "*"}
 spacy = {optional = true, version = "^3.7.1"}
 aiohttp = "^3.8.6"
 networkx = ">=3.0"
+<<<<<<< HEAD
+psycopg2 = {optional = true, version = "^2.9.9"}
```
[the remark this answers is not in the document]
Comment (logan-markewich) [reply pyproject.toml:81]: yea lets not edit the pyproject.toml -- will undo this change :)

File: llama_index/vector_stores/azureaisearch.py [hunk 5]
```diff
@@ -45,7 +45,7 @@ class IndexManagement(int, enum.Enum):
     CREATE_IF_NOT_EXISTS = auto()
 
 
-class CognitiveSearchVectorStore(VectorStore):
+class AzureAISearchVectorStore(VectorStore):
```
[the remark this answers is not in the document]
Comment (logan-markewich) [reply llama_index/vector_stores/azureaisearch.py:48]: I think at a minimum, putting `CognitiveSearchVectorStore = AzureAISearchVectorStore` at the bottom of the page may help, and keeping it in the `__init__.py` file?

The overall interface does not appear to have changed in a breaking way?

File: llama_index/vector_stores/azureaisearch.py [hunk 6]
```diff
@@ -141,68 +148,69 @@ def _create_index(self, index_name: Optional[str]) -> None:
...
+                ),
+            ],
+            profiles=[
+                VectorSearchProfile(
+                    name="myHnswProfile",
+                    algorithm_configuration_name="myHnsw",
+                ),
+                # Add more profiles if needed
+                VectorSearchProfile(
+                    name="myExhaustiveKnnProfile",
```
[the remark this answers is not in the document]
Comment (farzad528) [reply llama_index/vector_stores/azureaisearch.py:192]: @logan-markewich thoughts on exposing this as an enum (Exhasutive Knn vs HNSW, likely more in the future)? Should we default all vector indexes to HNSW (the most popular)?

[the remark this answers is not in the document]
Comment (logan-markewich) [reply llama_index/vector_stores/azureaisearch.py:192]: I think that makes sense. Defaulting to hnsw is the right thing to do as well

File: llama_index/vector_stores/azureaisearch.py (hunk 1 above)
Comment (farzad528) [reply llama_index/vector_stores/azureaisearch.py:153]: I'm OK with that

File: llama_index/vector_stores/__init__.py [hunk 7]
```diff
@@ -8,7 +8,7 @@
 from llama_index.vector_stores.cassandra import CassandraVectorStore
 from llama_index.vector_stores.chatgpt_plugin import ChatGPTRetrievalPluginClient
 from llama_index.vector_stores.chroma import ChromaVectorStore
-from llama_index.vector_stores.cogsearch import CognitiveSearchVectorStore
+from llama_index.vector_stores.azureaisearch import AzureAISearchVectorStore
```
Comment (logan-markewich) [inline llama_index/vector_stores/__init__.py:11]: We will also need to import CognitiveSearchVectorStore from azureaisearch

Either that, we should probably also update the failing test and make sure it works
`llama_index/tests/vector_stores/test_cogsearch.py`

You can run tests with
`pytests llama_index/tests/vector_stores/test_cogsearch.py`

You might need to do `pip install pytest` for it to work, if it complains about missing pytest

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**